application/Core/CLI/Interactive/InteractiveCommandLine.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Threading;
+using MORR.Shared.Utility;
+
+namespace MORR.Core.CLI.Interactive
+{
+    internal class InteractiveCommandLine
+    {
+        private const char exitCommand = 'x';
+        private const string startMessage = "Use 'x' and enter to stop the recording!";
+        private const string closingMessage = "Closing MORR. This may take some time!";
+
+        public void Launch()
+        {
+            var userThread = new Thread(Start);
```

Comment:
If you start another thread anyway, why the change from the blocking `GetMessage` to the non-blocking `PeekMessage`?